main/src/main.ts

```diff
@@ -144,6 +153,14 @@ export function createWindow(commandLineArguments, workingDirectory) {
         })
     })
 
+    ipcMain.on("Oni.started", (evt) => {
```

Comment:
Looking at this - I'm wondering, will this be called for _every_ window? In other words, if `createWindow` is called multiple times, will this event get registered multiple times, and send the `delayedEvent` multiple times to `mainWindow`?

I'm wondering if this should be `ipcMain.once` instead of `ipcMain.on` to protect against this. The existing `rebuild-menu` event might have the same issue, too, but it seems less critical. For this event, it seems like if we call `createWindow` with a delayed event, and then we call it again with no delayedEvent, the listener in the first `createWindow` call would send us that original `delayedEvent` again, to the new window.

Reply:
Ah @bryphe right on the money 👍 I had noticed this quirk yesterday and wasn't sure what it was and wracking my brain as to what to do did not realise there was a `once` method which fully resolves that so that delayed events are only called once (what I was getting was open file of a closed window click open new file which works but then close that window and click again and the same command would execute, using `once` however resolves that).